src/riak_kv_token_manager.erl

```diff
@@ -613,8 +631,31 @@ monitor_manager(Manager, MonitoredManagers) ->
             [{Manager, MgrRef}|MonitoredManagers]
     end.
 
--spec check_upstream(token_id(), manager_pid(), session_pid()) -> ok. 
-check_upstream(TokenID, RemoteManager, Session) ->
+-spec check_active(list({token_id(), granted_session()}), pid()) -> ok.
+check_active([], _Mgr) ->
+    ok;
+check_active([{_TokenID, {local, Session, _VL, _VC}}|Rest], Mgr) ->
+    case is_process_alive(Session) of
+        true ->
+            check_active(Rest, Mgr);
+        false ->
```

Comment:
Well... possibly better to just monitor this dead pid, because then we don't have to know exactly what format the monitor uses.... 
Not really important but it clarifies that we have a process that we lost the monitor for and now we repair that,

Reply:
Changed as suggested